lib/order.rb

```diff
@@ -0,0 +1,46 @@
+require_relative 'menu'
+
+class Order
+  attr_reader :menu
+  attr_reader :basket
+
+  ERROR = "Item not found or unavailable"
+
+  def initialize(menu = Menu.new)
+    @basket = []
+    @menu = menu.display
+  end
+
+  def take(dish, quantity = 1)
+    raise ERROR unless dish_exists?(dish)
+
+    @basket << [dish,  quantity]
+    repeat_order(dish, quantity)
+  end
+
+  def display
+    basket_with_prices.map { |arr|
+      "#{arr[0]} x #{arr[1]} = £#{arr[2] * arr[1]}"
+    }.join(", ")
+  end
+
+  def details
+    basket_with_prices
+  end
+  
+  private
+
+  def basket_with_prices
+    @basket.map { |arr|
+      [arr[0], arr[1], @menu[arr[0]]]
+    }
+  end
+
+  def repeat_order(dish, quantity)
```

Comment:
What is the purpose of `#repeat_order`? It sounds as if it repeats the order, as in places a duplicate order. But when I read the code, I can see the item, quantity, and that it was added to the basket.